app/controllers/hq/partners_controller.rb

```diff
@@ -1,71 +1,44 @@
 class Hq::PartnersController < ApplicationController
-  before_action :authenticate_admin_user!
+  before_action :authenticate_admin_user!,
+                :new_partner_facade
   layout 'application'
 
   def index
-    @partners= Partner.all
+    @partners = Partner.all
   end
 
   def new
-    @partner = Partner.new
-    load_associations
+    @partner_facade.build_partner
   end
   
   def create 
-    build_partner
-    save_partner or re_render 'new'
+    @partner_facade.build_partner(params[:partner])
```

Comment:
I did not try that, it might work.

My concern is that by moving everything in the facade, it will make a very clean controller and a fat facade. It is not solving anything.
I think this is what Nik is also saying, the facade should only offer a "portal" for partner object manipulation and leave the flash, redirect, etc to the controller. We don't want to change rails normal behavior.  ¿